File: Daily Test/Java/Form List Of Integers 2.12.2025.java
import java.util.*;
class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Deque<Integer> dq = new ArrayDeque<>();

        for (int i = 1; i <= n; i++) {
            int x = sc.nextInt();
            if (i % 2 == 1) dq.addFirst(x);
            else dq.addLast(x);
        }

        if (n % 2 == 0) {
            List<Integer> temp = new ArrayList<>(dq);
            Collections.reverse(temp);
            for (int v : temp) System.out.print(v + " ");
        } else {
            for (int v : dq) System.out.print(v + " ");
        }
    }
}

File: Daily Test/Python/Form List Of Integers 2.12.2025.py
from collections import deque

n = int(input())
arr = list(map(int, input().split()))
dq = deque()

for i in range(1, n + 1):
    x = arr[i - 1]
    if i % 2 == 1:
        dq.appendleft(x)
    else:
        dq.append(x)

if n % 2 == 0:
    dq.reverse()

print(*dq)